packages/compass/src/main/application.ts

```diff
@@ -70,15 +78,8 @@ class CompassApplication {
     app.commandLine.appendSwitch('ignore-gpu-blacklist', 'true');
   }
 
-  private static async setupAutoUpdate(): Promise<void> {
-    if (process.env.HADRON_ISOLATED !== 'true') {
-      // This is done asyncronously so that webpack can completely remove
-      // autoupdater from the application bundle during compilation
-      const { CompassAutoUpdateManager } = await import(
-        './auto-update-manager'
-      );
-      CompassAutoUpdateManager.init();
-    }
+  private static setupAutoUpdate(): void {
+    CompassAutoUpdateManager.init();
```

Comment:
Should we check for `networkTraffic` / `autoUpdates` here before enabling autoupdates in the application, or should a check be added in the `CompassAutoUpdateManager` itself? I think right now calling init here would just unconditionally enable autoupdates which we don't want in isolated as far as I can tell

Reply:
I think what we’re aiming for here with the `HADRON_ISOLATED` and `HADRON_READONLY` is to load the necessary code always, and only alter runtime behavior depending on the settings values (and settings changes). In particular, the situation after this PR is the same as in the regular edition with the `autoUpdates` option turned off – and if auto updates still behave as if they were enabled in that case, that’s a bug either way.

We only call the underlying Electron auto-updater here: https://github.com/mongodb-js/compass/blob/6ba206d37b0a9a51372bbccede1dd66efdfbb595/packages/hadron-auto-update-manager/index.js#L136

if `AutoUpdateManager.enable()` has been called here: https://github.com/mongodb-js/compass/blob/6ba206d37b0a9a51372bbccede1dd66efdfbb595/packages/hadron-auto-update-manager/index.js#L183

which in turn is called only from here: https://github.com/mongodb-js/compass/blob/6ba206d37b0a9a51372bbccede1dd66efdfbb595/packages/compass/src/main/auto-update-manager.ts#L77-L78

and finally the call site of that event is guarded by taking in the actual value of the `autoUpdates` preference.

We _are_ going to simplify this call tree in COMPASS-6107.

Reply:
Got it, thanks for context! Without this it was hard to figure out whether we are changing the behavior or not. It's good to know that we are going to address the flow here, I kinda wish we did these in a different order so that reviewing this change would be easier 😅